src/types/num.rs

```diff
@@ -243,12 +243,21 @@ mod slow_128bit_int_conversion {
 
             impl IntoPy<PyObject> for $rust_type {
                 fn into_py(self, py: Python) -> PyObject {
-                    let bytes = PyBytes::new(py, &self.to_le_bytes());
-                    let longtype = unsafe { PyType::from_type_ptr(py, &mut ffi::PyLong_Type) };
-                    longtype
-                        .call_method("from_bytes", (bytes, "little"), kwargs(py, $is_signed))
-                        .expect("Integer conversion (u128/i128 to PyLong) failed")
-                        .into_py(py)
+                    let (first, last) = split_128int(self.to_le_bytes());
```

Comment:
Can I suggest we use the terminology `lower` / `upper` instead of `first` / `last`? For me it's easier to understand which bits are most significant / least significant with that terminology.

```suggestion
                    let (lower, upper) = split_128int(self.to_le_bytes());
```

Reply:
Thanks, I wanted better names too 😉